deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -839,7 +844,11 @@ def post_add_stage(request, name):
         else:
             common.create_simple_stage(request,name, stage, description, external_id)
     except:
-        environs_helper.delete_nimbus_identifier(request, external_id)
+        try:
+            environs_helper.delete_nimbus_identifier(request, external_id)
+        except TeletraanException as detail:
+            log.error('Handling TeletraanException when trying to access nimbus API, error message {}'.format(detail))
+            messages.add_message(request, messages.ERROR, detail)     
```

Comment:
So are you saying that we do or do not want to display the message to the user? if the nimbus identifier was not created and `delete_nimbus_identifier` fails shouldn't we just log the error and process with the operation? Since there is nothing to delete  and you said that we shouldn't show that kind of error to the user we should get rid of the raise statements.

Reply:
Sorry for the confusion. My second thought is to show the instructive error message. Note that the inner exception is based on the outer exception, so both errors are returned to users.

Reply:
Got it. Thanks for clarifying and this should probably surfaced as a banner since we still want to maintain render the page